projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-v1-1-endpoint.php

```diff
@@ -63,33 +63,51 @@ public function callback( $path = '', $blog_id = 0, $post_id = 0 ) {
 
 		$args = $this->query_args();
 
-		$site = $this->get_platform()->get_site( $blog_id );
-
 		if ( str_contains( $path, '/posts/slug:' ) ) {
+			$site = $this->get_platform()->get_site( $blog_id );
+
 			$post_id = $site->get_post_id_by_name( $post_id );
 			if ( is_wp_error( $post_id ) ) {
 				return $post_id;
 			}
 		}
 
-		if ( defined( 'IS_WPCOM' ) && IS_WPCOM &&
-				! in_array( get_post_type( $post_id ), array( false, 'post', 'revision' ), true ) ) {
+		return $this->fetch_post( $blog_id, $post_id, $args['context'] );
+	}
+
+	/**
+	 * Helper function to fetch the content of a post. User validation
+	 * should be handled by the caller.
+	 *
+	 * @param int    $blog_id The blog ID for the post.
+	 * @param int    $post_id The post ID.
+	 * @param string $context The context we're fetching for.
+	 * @return array|SAL_Post|WP_Error
+	 */
+	public function fetch_post( $blog_id, $post_id, $context ) {
```

Comment:
Nit, the naming throughout is a little confusing; are we fetching the post or the post content? Maybe `fetch_post_content`/`$post_content` for clarity (or update the comment, either way).